be/src/formats/json/map_column.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "column/map_column.h"
+
+#include "fmt/format.h"
+#include "formats/json/map_column.h"
+#include "formats/json/nullable_column.h"
+#include "gutil/strings/substitute.h"
+
+namespace starrocks {
+
+Status add_map_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                      simdjson::ondemand::value* value) {
+    auto map_column = down_cast<MapColumn*>(column);
+
+    try {
+        simdjson::ondemand::object obj = value->get_object();
+        simdjson::ondemand::parser parser;
+        size_t field_count = 0;
+        for (auto field : obj) {
+            {
+                // This is a tricky way to transform a std::string to simdjson:ondemand:value
+                std::string_view field_name_str = field.unescaped_key();
+                auto dummy_json = simdjson::padded_string(R"({"dummy_key": ")" + std::string(field_name_str) + R"("})");
```

Comment:
What is the tricy way? Does the field_key need to be a JSON value rather than a simple string, so you parse a dummy_json to get the key? Why not implemet a add_nullable_column with string value?

Reply:
Yeah. Since the last argument of`add_nullable_column` is `simdjson::ondemand::value`, we have to convert the simple string to `simdjson::ondemand::value`.
There're many functions using the `simdjson::ondemand::value` as the input argument, implementing a `add_nullable_column` with string/float/double/int value will involve massive changes.